### DISCUSSION TITLE
using yield inside loop

### DISCUSSION THREAD

**Discussion by arashnz** [reactphp/reactphp#513] [Q&A]
hello,

how can i use yield keyword inside React\EventLoop\Loop::get() and new Connector($loop)?
it just don't work when i use yield inside it

**Comment by SimonFrings** [comment]
Hey @arashnz, thanks for bringing this up :+1:

I'm not sure if I completely understand the problem you're having, as a start I would suggest to take a look at the coroutine documentation in https://github.com/reactphp/async#

Taken from this documentation, here's an example of using `yield`:

```php
React\Async\coroutine(function () {
    $browser = new React\Http\Browser();

    try {
        $response = yield $browser->get('https://example.com/');
        assert($response instanceof Psr\Http\Message\ResponseInterface);
        echo $response->getBody();
    } catch (Exception $e) {
        echo 'Error: ' . $e->getMessage() . PHP_EOL;
    }
});
```

If this doesn't help you to progress, please come back to this conversation and provide some more information like:

- What do you want to achieve?
- What steps have you taken so far, what worked and what didn't?
- What does your code look like (code examples are always welcome)?
- If you see any error messages, what do they say?

By providing us with this information, we can better understand your problem and give you a helpful solution! :+1:
